analyses/org.eclipse.tracecompass.incubator.executioncomparision.ui/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/ui/ExecutionComparisonView.java

```diff
@@ -0,0 +1,1212 @@
+
+                Action statisticActionDur = createStatisticAction(Objects.requireNonNull(Messages.multipleDensityViewDuration));
+                new ActionContributionItem(statisticActionDur).fill(menu, -1);
+
+                Action statisticActionSelf = createStatisticAction(Objects.requireNonNull(Messages.multipleDensityViewSelfTime));
+                new ActionContributionItem(statisticActionSelf).fill(menu, -1);
+                return menu;
+
+            }
+        });
+
+        return Objects.requireNonNull(configureStatisticAction);
+    }
+
+    /**
+     * @param signal
+     *            the TmfCheckboxChangedSignal signal
+     */
+    @TmfSignalHandler
+    public void checkBoxUpdated(TmfCheckboxChangedSignal signal) {
+        try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINE, "MultiDensityView::CheckBoxUpdated")) {//$NON-NLS-1$
+            TmfComparisonFilteringUpdatedSignal rangUpdateSignal = null;
+
+            if (signal.getSource() == getChartViewerA()) {
+                fTraceListA.clear();
+                for (String name : signal.getTraceList()) {
+                    fTraceListA.add(name);
+                }
+
+                rangUpdateSignal = new TmfComparisonFilteringUpdatedSignal(this, null, fTraceListA, null);
+                TmfSignalManager.dispatchSignal(rangUpdateSignal);
+            }
+
+            if (signal.getSource() == getChartViewerB()) {
+                fTraceListB.clear();
+                for (String name : signal.getTraceList()) {
+                    fTraceListB.add(name);
+                }
+
+                rangUpdateSignal = new TmfComparisonFilteringUpdatedSignal(this, null, null, fTraceListB);
+                TmfSignalManager.dispatchSignal(rangUpdateSignal);
+
+            }
+            if (ftextQuery != null) {
+                ftextQuery.setText(makeQuery());
+            }
+            buildDifferetialFlameGraph();
+        }
+
+    }
+
+    @Override
+    public void handleCheckStateChangedEvent(@SuppressWarnings("null") Collection<ITmfTreeViewerEntry> entries) {
+        // do nothing
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        TmfSignalManager.deregister(this);
+
+        getChartViewerA().dispose();
+        getTmfViewerA().dispose();
+        getChartViewerB().dispose();
+        getTmfViewerB().dispose();
+
+        if (fXYViewerContainerA != null) {
+            fXYViewerContainerA.dispose();
+        }
+        if (fXYViewerContainerB != null) {
+            fXYViewerContainerB.dispose();
+        }
+        if (fContextService != null) {
+            fContextService.dispose();
+        }
+        if (fSashFormLeftChildA != null) {
+            fSashFormLeftChildA.dispose();
+        }
+        if (fSashFormLeftChildB != null) {
+            fSashFormLeftChildB.dispose();
+        }
+        if (fsashForm != null) {
+            fsashForm.dispose();
+        }
+    }
+
+    private TmfXYChartViewer getChartViewerA() {
```

Comment:
All of these private "accessors" should be deleted.

Reply:
Removing these private accessors and using the class fields directly causes issues since these fields are `@Nullable`. Making them `@NonNull` also causes issues since we can't instantiate them in the constructor. 

What do you think is the best way of proceeding ? I can see two options but maybe I'm missing something:
1 - Keep the private accessors
2 - Use `Objects.requireNonNull()`

Reply:
All of the @Nullable should be kept where necessary. I do not believe most of those attributes are necessary as most of them are accessed in the createPartControl method only. The methods should all be deleted and, in most methods the values should be passed as parameters.

For example, the methods createResetButtonA, and B are mostly duplicated code which uses attributes whereas it should be using only parameters and this fixes the nullable issue because in createPartControl, we know that those values are not null because they are assigned before. This makes the static analysis more straight forward and avoid errors. 

Also in signal handlers, the fact that the values are nullable should be handled individually as right now, even though there are no warnings we could have some race conditions which would throw errors. Hiding them under functions and requireNonNull is not a good way of fixing those.

Reply:
Done 👍